### PR TITLE
BDOG-1471 Upgrade to Play 28

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

File: test/uk/gov/hmrc/filetransmission/utils/TestHttpClient.scala [hunk 1]
```diff
@@ -18,17 +18,25 @@ package uk.gov.hmrc.filetransmission.utils
...
-import com.typesafe.config.Config
+import com.typesafe.config.{Config, ConfigFactory}
 import play.api.libs.ws.ahc.AhcWSClient
-import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.http.hooks.HttpHook
 import uk.gov.hmrc.play.http.ws.WSHttp
+import scala.collection.JavaConverters._
 
 class TestHttpClient extends HttpClient with WSHttp {
```
Comment (colin-lamed) [inline test/uk/gov/hmrc/filetransmission/utils/TestHttpClient.scala:28]: you can use `HttpClientSupport` trait from `http-verbs-test`
the configuration defaults should suffice too.

File: test/uk/gov/hmrc/filetransmission/services/queue/MongoBackedWorkItemServiceSpec.scala [hunk 2]
```diff
@@ -74,7 +74,7 @@ class MongoBackedWorkItemServiceSpec
         new MongoBackedWorkItemService(repository, job, configuration, clock)
 
       When("the request was enqueued")
-      Await.result(service.enqueue(envelope), 10 seconds)
+      Await.result(service.enqueue(envelope), 10.seconds)
```
Comment (colin-lamed) [inline test/uk/gov/hmrc/filetransmission/services/queue/MongoBackedWorkItemServiceSpec.scala:77]: using `ScalaFutures` would be nicer than all these `Await`s

File: app/uk/gov/hmrc/filetransmission/connector/HttpCallbackSender.scala [hunk 3]
```diff
@@ -37,6 +37,8 @@ class HttpCallbackSender @Inject()(httpClient: HttpClient)(implicit ec: Executio
 
   implicit val failureCallback: Writes[FailureCallback] = Json.writes[FailureCallback]
 
+  implicit val legacyRawReads: HttpReads[HttpResponse] = HttpReads.Implicits.throwOnFailure(HttpReads.Implicits.readEitherOf(HttpReads.Implicits.readRaw))
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/filetransmission/connector/HttpCallbackSender.scala:40]: We could rework the call to check the response code and not use the legacy raw reads.

**Review by colin-lamed — APPROVED** [review]

File: it/TransmissionRequestWorkItemRepositoryISpec.scala [hunk 4]
```diff
@@ -40,35 +42,38 @@ class TransmissionRequestWorkItemRepositoryISpec
     )
     .build()
 
+  override implicit val patienceConfig: PatienceConfig = PatienceConfig(
```
Comment (colin-lamed) [inline it/TransmissionRequestWorkItemRepositoryISpec.scala:45]: adding the trait `IntegrationPatience` is usually enough

**Review by colin-lamed — APPROVED** [review]

File: it/TransmissionRequestWorkItemRepositoryISpec.scala (hunk 4 above)
Comment (chotai) [reply it/TransmissionRequestWorkItemRepositoryISpec.scala:45]: Some of these tests had a 5 second timeout and others 10 so thought to preserve that behaviour

**Pull request merged**